manifests/install.pp

```diff
@@ -2,21 +2,24 @@
 #
 # @api private
 class qpid::install {
-  include qpid::tools
+
+  if $qpid::ensure == 'absent' {
+    $_package_ensure = 'purged'
+  } else {
+    $_package_ensure = $qpid::version
+  }
 
   package { $qpid::server_packages:
-    ensure => $qpid::version,
-    before => Class['qpid::tools'],
+    ensure => $_package_ensure,
   }
 
   if $qpid::auth {
-    ensure_packages(['cyrus-sasl-plain'])
+    ensure_packages(['cyrus-sasl-plain'], {ensure => $_package_ensure})
```

Comment:
I'm not sure if this can just be removed. It may be used by others. I'd consider changing the `if` above it to not manage it if it should be absent.

Reply:
I dont know of anything else that should be using this on a system with qpid and specifically in our standard use case for sure; however, I could go either way with how much we do or do not cleanup. Also, I don't understand "changing the if above it to not manage it" statement.

Reply:
I'd write:
```puppet
if $qpid::auth && $qpid::ensure == 'present' {
  ensure_pacakges(['cyrus-sasl-plain'])
}
```

Essentially, when you use `ensure_packages` you say that you need the package, but another module may also need it. This is commonly used for tooling. `cyrus-sasl`-plain` may be used in other places. I recall having set up a postfix server using SASL in the past. Those kind of setups are the ones you could break.

Reply:
Ok. Educate me if you will, why would you put your postfix server on the same box as your Qpid - Katello install?

Reply:
It was just an example. Postfix could also be installed as an email relay. Foreman by default uses `/usr/bin/sendmail` which is provided by Postfix on EL. I can imagine that some might harden it where SASL is involved. For example as a relay for isolated Smart Proxies that still need to send cron reports.

And you're right that in environments where the OS is free of cost you might deploy a separate machine for better isolation but if I was a RHEL customer and would need to pay money for it, I'd probably combine that functionality.

My main point is that it's about learning the right patterns. Using `ensure_packages` with ensuring absent is a hint you may be doing it wrong. There are probably exceptions, but it should always flag a reviewer because in most cases it's using a pattern the wrong way.